### PR TITLE
Make copies of Vectors in Values Operator

### PR DESCRIPTION
**Pull request by kevinwilfong**
Summary:
The Values Operator takes a std::vector of RowVectors and simply returns them, it's usually used as a source in test queries.

The std::vector of RowVectors comes from the Plan.  There's a risk of concurrency issues because the std::vector is shared 
across Drivers running on multiple threads.

This is demonstrated by the AssertQueryBuilderTest.concurrency test which currently fails with TSAN. It makes a copy of the
output RowVectors, which are simply the Vectors read from Values, which leads to data races as the field 
containsLazyNotLoaded_ gets updated by the concurrent copies.

Since the Drivers are created on a single thread doing the copies when the Operators are created is thread safe.  Since it's
only done once at Driver startup the cost of the copy is minimal.

Differential Revision: D53251319

### CODE REVIEW

File: velox/functions/prestosql/aggregates/tests/ApproxPercentileTest.cpp [hunk 1]
```diff
@@ -400,27 +401,63 @@ TEST_F(ApproxPercentileTest, finalAggregateAccuracy) {
...
-      ARRAY(DOUBLE()),
-      nullptr,
-      1,
-      AlignedBuffer::allocate<vector_size_t>(1, pool(), 0),
-      AlignedBuffer::allocate<vector_size_t>(1, pool(), 3),
-      BaseVector::wrapInDictionary(
-          nullptr, indices, 3, makeFlatVector<double>({0, 0.5, 1})));
+  exec::registerVectorFunction(
+      "testing_dictionary_percentiles",
+      TestingDictionaryPercentilesFunction::signatures(),
```
Comment (bikramSingh91) [inline velox/functions/prestosql/aggregates/tests/ApproxPercentileTest.cpp:450]: curious, why are we creating a UDF to generate the percentiles vector?

File: velox/parse/QueryPlanner.cpp [hunk 2]
```diff
@@ -108,15 +108,18 @@ PlanNodePtr toVeloxPlan(
...
+    children.emplace_back(
+        BaseVector::createNullConstant(types.back(), 1, pool));
   }
 
   auto rowType = ROW(std::move(names), std::move(types));
 
-  std::vector<RowVectorPtr> vectors = {std::make_shared<RowVector>(
-      pool, rowType, nullptr, 1, std::vector<VectorPtr>{})};
+  std::vector<RowVectorPtr> vectors = {
+      std::make_shared<RowVector>(pool, rowType, nullptr, 1, children)};
```
Comment (bikramSingh91) [inline velox/parse/QueryPlanner.cpp:122]: is this related to the fix? this looks like an invalid vector in general

File: velox/functions/prestosql/aggregates/tests/ApproxPercentileTest.cpp (hunk 1 above)
Comment (kevinwilfong) [reply velox/functions/prestosql/aggregates/tests/ApproxPercentileTest.cpp:450]: Making a copy in the ValuesOperator flattens the data, this test is trying to test what happens when percentiles is not flat.

Generating it in a UDF is a way to keep the encoding.

File: velox/parse/QueryPlanner.cpp (hunk 2 above)
Comment (kevinwilfong) [reply velox/parse/QueryPlanner.cpp:122]: It was a generally invalid vector, but the tests that produce a LogicalDummyScan in duckdb don't actually use this Vector.

E.g. this happens when you SELECT from an UNNEST expression instead of a table
SELECT x, x + 5 FROM UNNEST([1, 2, 3]) as t(x)

Here we want to read a row from somewhere (the ValuesOperator), but it's children are never accessed, it's enough that the vector simply exists.

With this change the ValuesOperator is copying the Vector, so now they are accessed.  We still don't care what's in it, but we do care that it's valid.

**Comment by kevinwilfong** [comment]
Discussed offline with @oerling and @bikramSingh91 and will take a different approach.
